### DISCUSSION TITLE
[Unstable] Handheld Sonar & Artifact Container lights stuck on in subeditor

### DISCUSSION THREAD

**Discussion by Ek-Videogames** [FakeFishGames/Barotrauma#14830] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Handheld sonar & Artifact container now draw a light by default, and even when inside a container.

This causes unnecessary light rendering, especially with many manually-placed handheld sonars in a diving gear locker for example.

### Reproduction steps

- Artifact containers can be fixed by turning off their light component manually.

- Handheld sonars however cannot, and will for some reason *magically re-enable their light* upon exiting and reloading the save.

![image](https://github.com/user-attachments/assets/cb39baba-a6ef-4c42-a821-3b8ebc64a8f6)

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens outside single player or multiplayer game modes (e.g. game launches on startup, something broken in the main menu)

### Version

v1.6.4.0 (unstable)

### Which operating system did you encounter this bug on?

Windows

**Comment by Ek-Videogames** [comment]
Jukebox apparently also has this issue, which makes it impossible to have it spawn 'turned off' at the beginning of the round. Even setting it's signal output to "false" and turning off it's light - the light component spawns enabled by default.

**Reply by Regalis11** [reply]
The state of the jukebox's light is based on whether the jukebox is on or not - it should be possible to disable it by turning the jukebox off.

**Comment by michael-fakefish** [comment]
Tested the fix for this issue. Closing the discussion as resolved.
